ArticleTemplates/assets/js/modules/youtube.js

```diff
@@ -0,0 +1,246 @@
+        }
+    }
+
+    function setupPlayer(id) {
+        var player = new YT.Player(id, {
+            events: {
+                onReady: onPlayerReady.bind(null, id),
+                onStateChange: onPlayerStateChange.bind(null, id)
+            }
+        });
+
+        return player;
+    }
+
+    function onPlayerReady(id) {
+        var touchPoint;
+
+        players[id].duration = players[id].player.getDuration();
+
+        if (players[id].placeholder) {
+            touchPoint = players[id].placeholder.getElementsByClassName('youtube-media__touchpoint')[0];
+            
+            if(!GU.opts.nativeYoutubeEnabled || GU.opts.nativeYoutubeEnabled !== 'true') {
+                touchPoint.addEventListener('click', playVideo.bind(null, id, players[id].placeholder.parentNode));
+            } else {
+                touchPoint.addEventListener('click', sendPlayEventForNativePlayer.bind(null, id));
+            }
+            
+            players[id].placeholder.classList.add('fade-touchpoint');
+        } else {
+            players[id].iframe.parentNode.classList.add('show-video');
+        }
+    }
+
+    function sendPlayEventForNativePlayer(id) {
+        trackEvent({
+            id: id,
+            eventType: 'video:content:start'
+        });
+    }
+
+    function playVideo(id, placeholderParent) {
+        players[id].player.playVideo();
+        placeholderParent.classList.add('show-video');
+        setTimeout(hidePlaceholder.bind(null, placeholderParent), 300);
+    }
+
+    function hidePlaceholder(placeholderParent) {
+        placeholderParent.classList.add('hide-placeholder');
+    }
+
+    function onPlayerStateChange(id, event) {
+        Object.keys(stateHandlers).forEach(checkState.bind(null, id, event.data));
+    }
+
+    function checkState(id, state, stateHandler) {
+        if (state === window.YT.PlayerState[stateHandler]) {
+            stateHandlers[stateHandler](id);
+        }
+    }
+
+    function onPlayerPlaying(id) {
+        var currentTime = Math.round(players[id].player.getCurrentTime());
+
+        stopPlayers(id);
+        setProgressTracker(id);
+
+        if (currentTime === 0) {
+            trackEvent({
+                id: id,
+                eventType: 'video:content:start'
+            });
+        }
+    }
+
+    function onPlayerEnded(id) {
+        var placeholderParent; 
+
+        if (players[id].placeholder) {
+            placeholderParent = players[id].placeholder.parentNode;
+            placeholderParent.classList.remove('hide-placeholder');
+            setTimeout(showPlaceholder.bind(null, placeholderParent), 1000);
+        }
+
+        killProgressTracker(false, id);
+
+        trackEvent({
+            id: id,
+            eventType: 'video:content:end'
+        });
+    }
+
+    function showPlaceholder(placeholderParent) {
+        placeholderParent.classList.remove('show-video');
+    }
+
+    function onPlayerPaused(id) {
+        killProgressTracker(false, id);
```

Comment:
@akash1810 just to prevent the tracker (polling) function from continuing to tick whilst the videos not playing.